fe/fe-core/src/main/java/org/apache/doris/analysis/Analyzer.java

```diff
@@ -553,7 +552,7 @@ public TableRef resolveTableRef(TableRef tableRef) throws AnalysisException {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_STATE, "RESTORING");
         }
 
-        TableName tblName = new TableName(database.getFullName(), table.getName());
+        TableName tblName = new TableName(dbName, tableName.getTbl());
```

Comment:
Add comment to explain why using `tableName.getTbl()` instead of `table.getName()` here.